crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -337,9 +403,227 @@ where
                         + &ValueSum::from_pair((*token).clone(), post_balance)
                 )
                 .map_err(native_vp::Error::new)?;
-                Ok(result)
+                Result::<_>::Ok(result)
             },
-        )
+        )?;
+
+        let ibc_address = MaybeIbcAddress::Address(IBC);
+        // Public keys must be the hash of the sources/targets
+        let ibc_address_hash = TransparentAddress(<[u8; 20]>::from(
+            ripemd::Ripemd160::digest(sha2::Sha256::digest(
+                &ibc_address.serialize_to_vec(),
+            )),
+        ));
+        // Enable decoding the IBC address hash
+        changed_balances.decoder.insert(ibc_address_hash, ibc_address);
+
+        // Extract the IBC events
+        let ibc_events: BTreeSet<_> = self
```

Comment:
We don't need IBC events because the transaction has only one IBC message.
We can get the IBC transfer info by decoding the tx_data like https://github.com/anoma/namada/blob/879a3268f2edfbdd0da1c98cf0e4bbdac48f5b0d/crates/ibc/src/lib.rs#L156.